unstructured_ingest/v2/processes/connectors/pinecone.py

```diff
@@ -159,18 +159,64 @@ def conform_dict(self, element_dict: dict, file_data: FileData) -> dict:
 
 
 @dataclass
-class PineconeUploader(Uploader):
+class PineconeUploader(VectorDBUploader):
     upload_config: PineconeUploaderConfig
     connection_config: PineconeConnectionConfig
     connector_type: str = CONNECTOR_TYPE
 
+    # TODO: question this
+    def init(self, **kwargs: Any) -> None:
+        self.create_destination()
+
+    def index_exists(self, index_name: Optional[str]) -> bool:
+        index_name = index_name or self.connection_config.index_name
+        pc = self.connection_config.get_client()
+        return index_name in [ind["name"] for ind in pc.list_indexes()]
+
     def precheck(self):
         try:
-            self.connection_config.get_index()
+            self.connection_config.get_client().list_indexes()
+            if self.connection_config.index_name and not self.index_exists():
+                raise DestinationConnectionError(
+                    f"index {self.connection_config.index_name} does not exist"
+                )
         except Exception as e:
             logger.error(f"failed to validate connection: {e}", exc_info=True)
             raise DestinationConnectionError(f"failed to validate connection: {e}")
 
+    def create_destination(
+        self,
+        vector_length: int,
+        # TODO: make name required to either be in config, or here, then remove default
+        destination_name: Optional[str] = "unstructured-pinecone-default-index",
+        cloud: Optional[str] = "aws",
+        region: Optional[str] = "us-east-1",
+        metric: Optional[str] = "cosine",
```

Comment:
I think these would be required if creating a new destination. Having the default will make sure of that. 
```suggestion
        cloud: str = "aws",
        region: str = "us-east-1",
        metric: str = "cosine",
```